src/components/KTableData/KTableData.vue

```diff
@@ -0,0 +1,706 @@
+    default: '',
+  },
+  emptyStateIconVariant: {
+    type: String as PropType<EmptyStateIconVariant>,
+    default: EmptyStateIconVariants.Default,
+  },
+  emptyStateButtonAppearance: {
+    type: String as PropType<ButtonAppearance>,
+    default: null,
+  },
+  error: {
+    type: Boolean,
+    default: false,
+  },
+  errorStateTitle: {
+    type: String,
+    default: 'An error occurred',
+  },
+  errorStateMessage: {
+    type: String,
+    default: 'Data cannot be displayed due to an error.',
+  },
+  errorStateActionRoute: {
+    type: [Object, String] as PropType<RouteLocationRaw | string>,
+    default: null,
+  },
+  errorStateActionMessage: {
+    type: String,
+    default: '',
+  },
+  headers: {
+    type: Array as PropType<TableDataHeader[]>,
+    default: () => [],
+  },
+  maxHeight: {
+    type: String,
+    default: 'none',
+  },
+  hidePagination: {
+    type: Boolean,
+    default: false,
+  },
+  paginationAttributes: {
+    type: Object as PropType<TablePaginationAttributes>,
+    default: () => ({}),
+  },
+  rowExpandable: {
+    type: Function as PropType<(row: Record<string, any>) => boolean>,
+    default: () => false,
+  },
+  hideHeaders: {
+    type: Boolean,
+    default: false,
+  },
+  nested: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * KTableData props
+   */
+  /**
+   * A prop to pass in a fetcher function to enable server-side search, sort
+   * and pagination
+   */
+  fetcher: {
+    type: Function,
+    default: undefined,
+    required: true,
+  },
+  /**
+   * A prop to trigger a revalidate of the fetcher function. Modifying this value
+   * will trigger a manual refetch of the table data.
+   */
+  fetcherCacheKey: {
+    type: String,
+    default: '',
+  },
+  /**
+   * A prop used to uniquely identify this table in the swrv cache
+   */
+  cacheIdentifier: {
+    type: String,
+    default: '',
+  },
+  /**
+   * A prop to pass in a search string for server-side search
+   */
+  searchInput: {
+    type: String,
+    default: '',
+  },
+  /**
+   * A prop to pass in an object of intial params for the initial fetcher function call
+   */
+  initialFetcherParams: {
+    type: Object,
+    default: null,
+  },
+  /**
+   * Enable client side sort - only do this if using a fetcher that returns static data
+   */
+  clientSort: {
+    type: Boolean,
+    default: false,
+  },
+  sortHandlerFunction: {
+    type: Function as PropType<(param: SortHandlerFunctionParam) => Record<string, any>[]>,
+    default: () => ({}),
+  },
+  sortable: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+   * A boolean to hide pagination when total table records number is less than or equal to page size
+   */
+  hidePaginationWhenOptional: {
+    type: Boolean,
+    default: false,
+  },
+  hideToolbar: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'cell-click', value: { data: any }): void
+  (e: 'row-click', value: { data: any }): void
```

Comment:
question: wouldn't the cell & row data at least conform to `Record<string, any>` rather than just `any`?